python/batch-predictor/merlinpyspark/__main__.py

```diff
@@ -36,16 +34,88 @@
 
 DEFAULT_PARALLELISM = 2
 
-
-def main(spec_path, spark):
+parser = argparse.ArgumentParser(description="Run a PySpark job")
+parser.add_argument(
+    "--job-name",
+    type=str,
+    required=False,
+    dest="job_name",
+    help="The name of PySpark job",
+    default="merlin-prediction-job",
+)
+parser.add_argument(
+    "--spec-path",
+    type=str,
+    required=False,
+    dest="spec_path",
+    help="Path to prediction job yaml file",
+)
+parser.add_argument(
+    "--dry-run-model",
+    type=str,
+    required=False,
+    dest="dry_run_path",
+    help="Path to model for dry run",
+    default=None,
+)
+parser.add_argument(
+    "--local",
+    dest="local",
+    action="store_true",
+    required=False,
+    help="flag to run locally",
+    default=False,
+)
+
+args, _ = parser.parse_known_args()
+print(f"Called with arguments: {args}")
```

Comment:
will this be executed on imports since these are in a global scope?

Reply:
correct. but it should be fine since it should not be imported. the similar approach is used by pyfuncserver.__main__